test/unit/power_manager_test.rb

```diff
@@ -0,0 +1,37 @@
+require 'test_helper'
+
+class PowerManagerTest < ActiveSupport::TestCase
+  test "should respond to all real actions with compute resource" do
+    vm_mock = mock('vm')
+
+    compute_resource_mock = mock('compute_resource')
+    compute_resource_mock.stubs(:find_vm_by_uuid).returns(vm_mock)
+
+    host = FactoryGirl.build(:host, :on_compute_resource)
+    host.unstub(:queue_compute)
+    host.stubs(:compute_resource).returns(compute_resource_mock)
+
+    PowerManager::SUPPORTED_ACTIONS.each do |action|
+      vm_mock.stubs(action.to_sym).returns(true)
+    end
+
+    PowerManager::REAL_ACTIONS.each do |action|
```

Comment:
I don't know that this is the right constant.  I think it should be SUPPORTED_ACTIONS, which appears to be the list of everything the power manage is expected to respond to.